flow-renderer-parent/flow-renderer/src/main/resources/META-INF/resources/frontend/lit-renderer.ts

```diff
@@ -0,0 +1,108 @@
+ * @param clientCallables A list of function names that can be called from within the template literal.
+ * @param propertyNamespace LitRenderer-specific namespace for properties.
+ * Needed to avoid property name collisions between renderers.
+ */
+_window.Vaadin.setLitRenderer = (
+  component: HTMLElement,
+  rendererName: string,
+  templateExpression: string,
+  returnChannel: (name: string, itemKey: string, args: any[]) => void,
+  clientCallables: string[],
+  propertyNamespace: string
+) => {
+  // Dynamically created function that renders the templateExpression
+  // inside the given root element using Lit
+  const renderFunction = Function(`
+    "use strict";
+
+    const [render, html, returnChannel] = arguments;
+
+    return (root, {item, index}, itemKey) => {
+      ${clientCallables
+        .map((clientCallable) => {
+          // Map all the client-callables as inline functions so they can be accessed from the template literal
+          return `
+          const ${clientCallable} = (...args) => {
+            if (itemKey !== undefined) {
+              returnChannel('${clientCallable}', itemKey, args[0] instanceof Event ? [] : [...args]);
+            }
+          }`;
+        })
+        .join('')}
+
+      render(html\`${templateExpression}\`, root)
+    }
+  `)(render, html, returnChannel);
+
+  const renderer: Renderer = (root, _, { index, item }) => {
+    // Clean up the root element of any existing content
+    // (and Lit's _$litPart$ property) from other renderers
+    // TODO: Remove once https://github.com/vaadin/web-components/issues/2161 is done
+    if (root.__litRenderer !== renderer) {
+      root.innerHTML = '';
+      delete root._$litPart$;
+      root.__litRenderer = renderer;
+    }
+
+    // Map a new item that only includes the properties defined by
+    // this specific LitRenderer instance. The renderer instance specific
+    // "propertyNamespace" prefix is stripped from the property name at this point:
+    //
+    // item: { key: "2", lr_3_lastName: "Tyler"}
+    // ->
+    // mappedItem: { lastName: "Tyler" }
+    const mappedItem = {};
+    for (const key in item) {
+      if (key.startsWith(propertyNamespace)) {
+        mappedItem[key.replace(propertyNamespace, '')] = item[key];
+      }
+    }
+
+    renderFunction(root, { index, item: mappedItem }, item.key);
+  };
+
+  (renderer as any).__rendererId = propertyNamespace;
+  component[rendererName] = renderer;
+};
+
+/**
+ * Removes the renderer function with the given name from the component
+ * if the propertyNamespace matches the renderer's id.
+ *
+ * @param component The host component whose renderer runction is to be removed
+ * @param rendererName The name of the renderer function
+ * @param rendererId The rendererId of the function to be removed
+ */
+_window.Vaadin.unsetLitRenderer = (
+  component: HTMLElement,
+  rendererName: string,
+  rendererId: string
+) => {
+  if (component[rendererName]?.__rendererId === rendererId) {
```

Comment:
The check for `__rendererId` property is necessary since the `renderer` function may get overridden by another renderer, for example, by one coming from `vaadin-template-renderer`. We don't want LitRenderer registration cleanup to unintentionally remove the new renderer.